js/index.js

```diff
@@ -28,36 +24,60 @@ messageForm.addEventListener('submit', e => {
     const messageList = messageSection.querySelector('ul');
     const newMessage = document.createElement('li');
 
-    newMessage.innerHTML = `<a href='mailto:${usersEmail}'>${usersName} </a> <span>wrote:</span> <span id="message">${usersMessage}</span>`;
+    newMessage.innerHTML = `<a href='mailto:${usersEmail}'>${usersName} </a> <span>wrote:</span> <div id="message">${usersMessage}</div>`;
 
     const removeButton = document.createElement('button');
     removeButton.innerText = 'remove';
     removeButton.type = 'button';
     const hideMessageSection = () => {
         messageList.children.length !== 0 ? messageSection.style.display = 'block' : messageSection.style.display = 'none'
     }
-    removeButton.addEventListener('click', e => {
+    const removeFunction = () => {
         const entry = removeButton.parentNode;
         entry.remove();
+    }
+    removeButton.addEventListener('click', e => {
+        removeFunction();
         hideMessageSection();
     })
 
+    const saveButton = document.createElement('button');
+    saveButton.innerText = 'save';
+    saveButton.type = 'button';
+    saveButton.style.display = 'none'
+    const saveFunction = () => {
+        saveButton.style.display = 'none'
+        editButton.style.display = 'inline';
+        removeButton.style.display = 'inline';
+        let entry = saveButton.parentNode.querySelector('#message');
+        let textarea = document.getElementById('messageTextarea');
+        entry.textContent = textarea.value;
+        textarea.replaceWith(entry);
+    }
+    saveButton.addEventListener('click', e => {
+        saveFunction();
+    })
+
     const editButton = document.createElement('button');
     editButton.innerText = 'edit';
     editButton.type = 'button';
-    editButton.addEventListener('click', e => {
-        const textarea = document.createElement('textarea');
-        const entry = editButton.parentNode.querySelector('#message');
-        entry.replaceWith(textarea)
+    const editFunction = () => {
+        editButton.style.display = 'none';
+        removeButton.style.display = 'none';
+        saveButton.style.display = 'inline'
+        let textarea = document.createElement('textarea');
+        textarea.id = 'messageTextarea';
+        let entry = editButton.parentNode.querySelector('#message');
+        entry.replaceWith(textarea);
```

Comment:
This is the bug here.  Before there is a div with id #message.  After, there is a textarea with id #messageTextArea, so the query for #message fails.